server/kit/log.go

```diff
@@ -104,23 +105,58 @@ func LogMsg(ctx context.Context, message string) error {
 	return level.Info(Logger(ctx)).Log("message", message)
 }
 
+// Logf will format the given string with the arguments then log to the server logger
+// with the key "message" along with all the common request headers or gRPC metadata.
+// Arguments are handlered in the manner of fmt.Printf.
+// This message will have an "info" log level associated with it.
+func Logf(ctx context.Context, format string, v ...interface{}) error {
+	return level.Info(Logger(ctx)).Log("message", fmt.Sprintf(format, v...))
+}
+
 // LogDebug will log the given message to the server logger
 // with the key "message" along with all the common request headers or gRPC metadata.
 // This message will have a "debug" log level associated with it.
 func LogDebug(ctx context.Context, message string) error {
 	return level.Debug(Logger(ctx)).Log("message", message)
 }
 
+// LogDebugf will format the given string with the arguments then log to the server
+// logger with the key "message" along with all the common request headers or gRPC
+// metadata.
+// Arguments are handlered in the manner of fmt.Printf.
+// This message will have a "debug" log level associated with it.
+func LogDebugf(ctx context.Context, format string, v ...interface{}) error {
```

Comment:
Do these need to return `error`?  If everyone will ignore it, what's the point?

Reply:
It's best practice in Go not to ignore errors and to either log them or return them. This lets the user decide how they want to handle (or ignore) the error.

Reply:
I get that, but in practice who's going to check error return on every logging stmt?  And then what do you do if it fails.

This is partially driven by IntelliJ flagging it when you don't check an error return.  I'm cool if you leave it, just wanted to point out that probably no-one will actually check the return values.

Reply:
That's correct, which is another one of the reasons I generally don't use those 3rd party metalinters. They go beyond the language specifications when it comes to expectations and can sometimes make code less readable.

Returning this error does at least allow the user to check if an error occurred, at least. If someone is having issues (esp around logging), this would at least be an option to be able to debug a problem without having to alter the gizmo library via gohack or some other tool.

Reply:
Those errors originate from the kit/log.Logger interface https://github.com/go-kit/kit/blob/master/log/log.go#L11